docs/Permissions/Onchain-Permissioning.md

```diff
@@ -0,0 +1,191 @@
+    Pantheon supports onchain permissioning for nodes. Onchain permissioning for accounts will be available 
+    in a future Pantheon release. 
+
+Onchain permissioning uses the following smart contracts: 
+
+* Ingress - a simple contract functioning as a gateway to the Rules contract. The Ingress contract is deployed 
+to a static address. 
+
+* Rules - stores the node whitelist and list of admin accounts. 
+
+## Pre-requisites 
+
+For nodes that are going to set up or maintain the nodes whitelist: 
+
+* [NodeJS](https://nodejs.org/en/) v8.9.4 or later 
+
+* [Truffle](https://truffleframework.com/docs/truffle/getting-started/installation)
+
+## Add Ingress Contract to Genesis File
+
+Add the Ingress contract to the genesis file for your network by copying it from [`genesis.json`](https://github.com/PegaSysEng/permissioning-smart-contracts/blob/master/genesis.json) 
+in the [`permissioning-smart-contracts` repository](https://github.com/PegaSysEng/permissioning-smart-contracts): 
+   
+```json
+"0x0000000000000000000000000000000000009999": {
+      "comment": "Ingress smart contract",
+      "balance": "0",
+      "code": <stripped>,
+      "storage": {
+         <stripped>
+      }
+}
+```
+
+!!! important 
+    Ensure your genesis file includes a sufficient milestone to support the permissioning contracts. 
+    For example, `constantinopleFixBlock`. 
+    
+## Onchain Permissioning Setup
+   
+1. Start your Pantheon node with contract permissioning options: 
+
+    * [--permissions-nodes-contract-enabled](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-enabled)
+      to enable onhcain permissioning
+
+    * [--permissions-nodes-contract-address](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-address)
+      set to the address of the Ingress contract in the genesis file (`"0x0000000000000000000000000000000000009999"`)
+
+1. Create the following environment variables and set to the specified values: 
+
+    * `PANTHEON_NODE_PERM_ACCOUNT` - address of account used to interact with the permissioning contracts. 
+
+    * `PANTHEON_NODE_PERM_KEY` - private key of the account used to interact with the permissioning contracts.
+
+    * `INGRESS_CONTRACT_ADDRESS` - address of the Ingress contract in the genesis file.  
+
+    * `PANTHEON_NODE_PERM_ENDPOINT` - required only if your node is not using the default JSON-RPC host and port (`http://127.0.0.1:8545`). 
+       Set to JSON-RPC host and port. 
+
+    !!! note
+        If your network is not configured as a [free gas network](../Configuring-Pantheon/FreeGas.md), the account used to 
+        interact with the permissioning contracts must have a balance. 
+
+1. Clone the `permissioning-smart-contracts` repository: 
+
+    ```bash 
+    git clone https://github.com/PegaSysEng/permissioning-smart-contracts.git
+    ```
+
+1. Change into the `permissioning-smart-contracts` directory and run: 
+
+    ```bash
+    npm install
+    ```
+
+## Deploy Rules Contract 
+
```

Comment:
make it clearer that only one node does this action. Ingress is shared at a specific location because it's in the genesis file, and the Rules contract is deployed (and registered with the Ingress contract) using truffle migrate. Only one node does this (there's nothing stopping you doing it a second time but then each truffle console will be talking to a different Rules contract).

Reply:
Actually, what will stop you from doing it a second time is - whether the account you're using is an admin account.

Reply:
Added an 'important' to make this clearer